tests/test_schedulers.py

```diff
@@ -0,0 +1,360 @@
+        for i in range(1, len(diffs)):
+            assert abs(diffs[i]) > abs(diffs[i - 1])
+
+
+class TestSigmaShifting:
+    """Tests for the sigma shifting function used for high-resolution images."""
+
+    @pytest.fixture
+    def resolutions(self):
+        """Resolutions to test with."""
+        return [
+            (512, 512),  # Smaller resolution
+            (1024, 1024),  # Medium resolution
+            (2048, 2048),  # Larger resolution
+        ]
+
+    def test_shift_shape_and_type(self, num_inference_steps, resolutions):
+        """Test the shape and type of shifted sigmas output."""
+        sigmas = RuntimeConfig._create_linear_sigmas(num_inference_steps)
+
+        for width, height in resolutions:
+            shifted_sigmas = RuntimeConfig._shift_sigmas(sigmas=sigmas, width=width, height=height)
+
+            # Check shape and type
+            assert shifted_sigmas.shape == sigmas.shape
+            assert shifted_sigmas.dtype == sigmas.dtype
+
+    def test_shift_boundary_conditions(self, num_inference_steps, resolutions):
+        """Test boundary conditions for shifted sigmas."""
+        sigmas = RuntimeConfig._create_linear_sigmas(num_inference_steps)
+
+        for width, height in resolutions:
+            shifted_sigmas = RuntimeConfig._shift_sigmas(sigmas=sigmas, width=width, height=height)
+
+            # Check that the last value is still 0
+            assert shifted_sigmas[-1] == 0.0
+
+            # Check that values are between 0 and 1
+            shifted_np = np.array(shifted_sigmas)
+            assert np.all(shifted_np >= 0.0)
+            assert np.all(shifted_np <= 1.0)
+
+    def test_shift_increases_with_resolution(self, num_inference_steps, resolutions):
+        """Test that higher resolutions produce higher shift values."""
+        sigmas = RuntimeConfig._create_linear_sigmas(num_inference_steps)
+
+        for width, height in resolutions:
+            if width > 512 and height > 512:
+                shifted_sigmas = RuntimeConfig._shift_sigmas(sigmas=sigmas, width=width, height=height)
+
+                # For most points, the shifted value should be higher than the original
+                # (but not necessarily all, and we need to exclude the 0 at the end)
+                shifted_np = np.array(shifted_sigmas)
+                orig_np = np.array(sigmas)[:-1]
+                shift_np = shifted_np[:-1]
+                assert np.sum(shift_np > orig_np) > len(orig_np) // 2
+
+
+class TestSchedulerSelection:
+    """Tests for the scheduler selection based on config."""
+
+    @pytest.fixture
+    def schedulers(self):
+        """List of scheduler types to test."""
+        return ["linear", "cosine", "exponential", "sqrt", "scaled_linear"]
+
+    def test_scheduler_selection(self, num_inference_steps, rtol, atol, model_config, schedulers):
+        """Test that _create_sigmas selects the correct scheduler based on the config."""
+        for scheduler in schedulers:
```

Comment:
Using `parametrize` would make it easier to e.g. `pytest -k cosine` to just test cosine stuff, and pytest output would also be cleaner (since each scheduler's test is separate there and you don't need to figure out where in the loop over schedulers it might be failing).
```suggestion
    @pytest.mark.parametrize("scheduler", NoiseSchedulerType.choices())
    def test_scheduler_selection(self, num_inference_steps, rtol, atol, model_config, scheduler):
            """Test that _create_sigmas selects the correct scheduler based on the config."""
```

Reply:
(That said, this test seems a bit pointless, since it's just basically copy-pasting the body of `_create_sigmas`?)